src/Symfony/Component/Cache/Adapter/FilesystemTagAwareAdapter.php

```diff
@@ -53,7 +53,7 @@ public function __construct(string $namespace = '', int $defaultLifetime = 0, st
     /**
      * {@inheritdoc}
      */
-    protected function doSave(array $values, ?int $lifetime, array $addTagData = [], array $removeTagData = []): array
+    protected function doSave(array $values, int $lifetime, array $addTagData = [], array $removeTagData = []): array
```

Comment:
The parent has an `int $lifetime` and it does not seem to ever be called with null as lifetime. Also current implementation would be broken with null because two lines below it calls `$this->doSaveCache($values, $lifetime);` which requires a sting as lifetime.